src/Interpreters/AsynchronousMetricLog.h

```diff
@@ -49,7 +51,7 @@ class AsynchronousMetricLog : public SystemLog<AsynchronousMetricLogElement>
     void addValues(const AsynchronousMetricValues &);
 
     /// This table is usually queried for fixed metric name.
-    static const char * getDefaultOrderBy() { return "metric, event_date, event_time"; }
+    static const char * getDefaultOrderBy() { return "hostname, metric, event_date, event_time"; }
```

Comment:
Not sure about this.

Reply:
It looks like this is needed. When I run the integration test locally, it indicates that there is a segfault that happens due to column mismatch without this:
```
2023.11.15 02:38:54.545868 [ 252 ] {} <Fatal> : Logical error: 'Invalid number of columns in chunk pushed to OutputPort. Expected 4, found 5
Header: event_date Date UInt16(size = 0), event_time DateTime UInt32(size = 0), metric LowCardinality(String) ColumnLowCardinality(size = 0, UInt8(size = 0), ColumnUnique(size = 1, String(size = 1))), value Float64 Float64(size = 0)
Chunk:  ColumnLowCardinality(size = 2360, UInt8(size = 2360), ColumnUnique(size = 2, String(size = 2))) UInt16(size = 2360) UInt32(size = 2360) ColumnLowCardinality(size = 2360, UInt16(size = 2360), ColumnUnique(size = 374, String(size = 374))) Float64(size = 2360)
'.
2023.11.15 02:38:54.547176 [ 276 ] {} <Fatal> BaseDaemon: ########## Short fault info ############
2023.11.15 02:38:54.547394 [ 276 ] {} <Fatal> BaseDaemon: (version 23.11.1.1, build id: 1F7F5CA22F8163637736DBB9DDCF5E1352AB160A, git hash: 4f677132514edae42ebe538ad242f38d73ac74c5) (from thread 252) Received signal 6
2023.11.15 02:38:54.547518 [ 276 ] {} <Fatal> BaseDaemon: Signal description: Aborted
```

Reply:
@bharatnc, It is assertion failure, not a "segfault".
Assertion failure is a memory-safe problem, similar to "panic" in Rust, while "segfault" can be either memory-safe (nullptr dereference) or memory unsafe.

Reply:
And we have to know the reason before merging.

Reply:
And you forgot to attach a stack trace.

Reply:
@alexey-milovidov thanks, that makes sense. I meant to use the word "crash". I will attach the stack trace shortly.

Reply:
```
2023.11.15 02:38:54.437484 [ 252 ] {} <Trace> SystemLog (system.asynchronous_metric_log): Flushing system log, 2360 entries to flush up to offset 2360
2023.11.15 02:38:54.437732 [ 252 ] {} <Debug> SystemLog (system.asynchronous_metric_log): Creating new table system.asynchronous_metric_log for AsynchronousMetricLog
2023.11.15 02:38:54.468529 [ 252 ] {} <Debug> system.asynchronous_metric_log (6769929f-56b8-41b5-a60b-c766a01465a6): Loading data parts
2023.11.15 02:38:54.468983 [ 252 ] {} <Test> system.asynchronous_metric_log (6769929f-56b8-41b5-a60b-c766a01465a6): loadDataParts: clearing data_parts_indexes (had 0 parts)
2023.11.15 02:38:54.469096 [ 252 ] {} <Debug> system.asynchronous_metric_log (6769929f-56b8-41b5-a60b-c766a01465a6): There are no data parts
2023.11.15 02:38:54.545868 [ 252 ] {} <Fatal> : Logical error: 'Invalid number of columns in chunk pushed to OutputPort. Expected 4, found 5
Header: event_date Date UInt16(size = 0), event_time DateTime UInt32(size = 0), metric LowCardinality(String) ColumnLowCardinality(size = 0, UInt8(size = 0), ColumnUnique(size = 1, String(size = 1))), value Float64 Float64(size = 0)
Chunk:  ColumnLowCardinality(size = 2360, UInt8(size = 2360), ColumnUnique(size = 2, String(size = 2))) UInt16(size = 2360) UInt32(size = 2360) ColumnLowCardinality(size = 2360, UInt16(size = 2360), ColumnUnique(size = 374, String(size = 374))) Float64(size = 2360)
'.
2023.11.15 02:38:54.546358 [ 7 ] {} <Trace> BaseDaemon: Received signal 6
2023.11.15 02:38:54.547176 [ 276 ] {} <Fatal> BaseDaemon: ########## Short fault info ############
2023.11.15 02:38:54.547394 [ 276 ] {} <Fatal> BaseDaemon: (version 23.11.1.1, build id: 1F7F5CA22F8163637736DBB9DDCF5E1352AB160A, git hash: 4f677132514edae42ebe538ad242f38d73ac74c5) (from thread 252) Received signal 6
2023.11.15 02:38:54.547518 [ 276 ] {} <Fatal> BaseDaemon: Signal description: Aborted
2023.11.15 02:38:54.547599 [ 276 ] {} <Fatal> BaseDaemon: 
2023.11.15 02:38:54.547736 [ 276 ] {} <Fatal> BaseDaemon: Stack trace: 0x00007f6970d989fc 0x00007f6970d44476 0x00007f6970d2a7f3 0x000000000fd5017f 0x000000000fd50215 0x000000000fd506d1 0x0000000007c63fea 0x00000000184c7984 0x0000000019967303 0x00000000199afdd9 0x00000000100964be 0x000000001998e6fc 0x000000001998511f 0x0000000019984b7e 0x00000000199aeeeb 0x00000000199aefd9 0x00000000182d69bb 0x00000000182d596e 0x000000000ff42099 0x000000000ff42075 0x000000000ff42059 0x000000000ff41fbd 0x000000000ff41ee5 0x000000000ff41e55 0x000000000ff41e35 0x000000000ff41e15 0x000000000ff41de0 0x000000000fdbddf6 0x000000000fdbd2d5 0x000000000fec5a23 0x000000000fecdcc4 0x000000000fecdc75 0x000000000fecdb9d 0x000000000fecd682 0x00007f6970d96ac3 0x00007f6970e28a40
2023.11.15 02:38:54.547846 [ 276 ] {} <Fatal> BaseDaemon: ########################################
2023.11.15 02:38:54.547918 [ 276 ] {} <Fatal> BaseDaemon: (version 23.11.1.1, build id: 1F7F5CA22F8163637736DBB9DDCF5E1352AB160A, git hash: 4f677132514edae42ebe538ad242f38d73ac74c5) (from thread 252) (no query) Received signal Aborted (6)
2023.11.15 02:38:54.548017 [ 276 ] {} <Fatal> BaseDaemon: 
2023.11.15 02:38:54.548078 [ 276 ] {} <Fatal> BaseDaemon: Stack trace: 0x00007f6970d989fc 0x00007f6970d44476 0x00007f6970d2a7f3 0x000000000fd5017f 0x000000000fd50215 0x000000000fd506d1 0x0000000007c63fea 0x00000000184c7984 0x0000000019967303 0x00000000199afdd9 0x00000000100964be 0x000000001998e6fc 0x000000001998511f 0x0000000019984b7e 0x00000000199aeeeb 0x00000000199aefd9 0x00000000182d69bb 0x00000000182d596e 0x000000000ff42099 0x000000000ff42075 0x000000000ff42059 0x000000000ff41fbd 0x000000000ff41ee5 0x000000000ff41e55 0x000000000ff41e35 0x000000000ff41e15 0x000000000ff41de0 0x000000000fdbddf6 0x000000000fdbd2d5 0x000000000fec5a23 0x000000000fecdcc4 0x000000000fecdc75 0x000000000fecdb9d 0x000000000fecd682 0x00007f6970d96ac3 0x00007f6970e28a40
2023.11.15 02:38:54.548206 [ 276 ] {} <Fatal> BaseDaemon: 4. ? @ 0x00007f6970d989fc in ?
2023.11.15 02:38:54.548300 [ 276 ] {} <Fatal> BaseDaemon: 5. ? @ 0x00007f6970d44476 in ?
2023.11.15 02:38:54.548373 [ 276 ] {} <Fatal> BaseDaemon: 6. ? @ 0x00007f6970d2a7f3 in ?
2023.11.15 02:38:54.667140 [ 276 ] {} <Fatal> BaseDaemon: 7. /home/bharatnc/Code/ClickHouse/src/Common/Exception.cpp:0: DB::abortOnFailedAssertion(String const&) @ 0x000000000fd5017f in /usr/bin/clickhouse
2023.11.15 02:38:54.782152 [ 276 ] {} <Fatal> BaseDaemon: 8. /home/bharatnc/Code/ClickHouse/src/Common/Exception.cpp:64: DB::handle_error_code(String const&, int, bool, std::vector<void*, std::allocator<void*>> const&) @ 0x000000000fd50215 in /usr/bin/clickhouse
2023.11.15 02:38:54.882198 [ 276 ] {} <Fatal> BaseDaemon: 9. /home/bharatnc/Code/ClickHouse/src/Common/Exception.cpp:101: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000fd506d1 in /usr/bin/clickhouse
2023.11.15 02:38:54.992105 [ 86 ] {} <Trace> test.test_table (fabdf8d3-2cf6-4ba5-b8aa-b60fdfa43437): Scheduling next merge selecting task after 7802ms
2023.11.15 02:38:54.996870 [ 276 ] {} <Fatal> BaseDaemon: 10. /home/bharatnc/Code/ClickHouse/src/Common/Exception.h:85: DB::Exception::Exception(String&&, int, bool) @ 0x0000000007c63fea in /usr/bin/clickhouse
2023.11.15 02:38:55.002621 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 588.16 MiB, peak 596.97 MiB, free memory in arenas 37.23 MiB, will set to 656.97 MiB (RSS), difference: 68.80 MiB
2023.11.15 02:38:55.312126 [ 276 ] {} <Fatal> BaseDaemon: 11. /home/bharatnc/Code/ClickHouse/src/Common/Exception.h:113: DB::Exception::Exception<unsigned long, unsigned long, String, String>(int, FormatStringHelperImpl<std::type_identity<unsigned long>::type, std::type_identity<unsigned long>::type, std::type_identity<String>::type, std::type_identity<String>::type>, unsigned long&&, unsigned long&&, String&&, String&&) @ 0x00000000184c7984 in /usr/bin/clickhouse
2023.11.15 02:38:55.399169 [ 276 ] {} <Fatal> BaseDaemon: 12.1. inlined from /home/bharatnc/Code/ClickHouse/src/Processors/Port.h:415: DB::OutputPort::pushData(DB::Port::State::Data)
2023.11.15 02:38:55.399326 [ 276 ] {} <Fatal> BaseDaemon: 12. /home/bharatnc/Code/ClickHouse/src/Processors/ISource.cpp:40: DB::ISource::prepare() @ 0x0000000019967303 in /usr/bin/clickhouse
2023.11.15 02:38:55.484314 [ 276 ] {} <Fatal> BaseDaemon: 13. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PushingPipelineExecutor.cpp:36: DB::PushingSource::prepare() @ 0x00000000199afdd9 in /usr/bin/clickhouse
2023.11.15 02:38:56.004014 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 656.97 MiB, peak 656.97 MiB, free memory in arenas 37.13 MiB, will set to 766.09 MiB (RSS), difference: 109.13 MiB
2023.11.15 02:38:56.023950 [ 276 ] {} <Fatal> BaseDaemon: 14. /home/bharatnc/Code/ClickHouse/src/Processors/IProcessor.h:193: DB::IProcessor::prepare(std::vector<unsigned long, std::allocator<unsigned long>> const&, std::vector<unsigned long, std::allocator<unsigned long>> const&) @ 0x00000000100964be in /usr/bin/clickhouse
2023.11.15 02:38:56.149883 [ 276 ] {} <Fatal> BaseDaemon: 15. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/ExecutingGraph.cpp:283: DB::ExecutingGraph::updateNode(unsigned long, std::queue<DB::ExecutingGraph::Node*, std::deque<DB::ExecutingGraph::Node*, std::allocator<DB::ExecutingGraph::Node*>>>&, std::queue<DB::ExecutingGraph::Node*, std::deque<DB::ExecutingGraph::Node*, std::allocator<DB::ExecutingGraph::Node*>>>&) @ 0x000000001998e6fc in /usr/bin/clickhouse
2023.11.15 02:38:56.286972 [ 276 ] {} <Fatal> BaseDaemon: 16. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PipelineExecutor.cpp:291: DB::PipelineExecutor::executeStepImpl(unsigned long, std::atomic<bool>*) @ 0x000000001998511f in /usr/bin/clickhouse
2023.11.15 02:38:56.418039 [ 276 ] {} <Fatal> BaseDaemon: 17. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PipelineExecutor.cpp:147: DB::PipelineExecutor::executeStep(std::atomic<bool>*) @ 0x0000000019984b7e in /usr/bin/clickhouse
2023.11.15 02:38:56.499182 [ 276 ] {} <Fatal> BaseDaemon: 18. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PushingPipelineExecutor.cpp:105: DB::PushingPipelineExecutor::push(DB::Chunk) @ 0x00000000199aeeeb in /usr/bin/clickhouse
2023.11.15 02:38:56.580706 [ 276 ] {} <Fatal> BaseDaemon: 19. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PushingPipelineExecutor.cpp:112: DB::PushingPipelineExecutor::push(DB::Block) @ 0x00000000199aefd9 in /usr/bin/clickhouse
2023.11.15 02:38:56.940844 [ 276 ] {} <Fatal> BaseDaemon: 20. /home/bharatnc/Code/ClickHouse/src/Interpreters/SystemLog.cpp:515: DB::SystemLog<DB::AsynchronousMetricLogElement>::flushImpl(std::vector<DB::AsynchronousMetricLogElement, std::allocator<DB::AsynchronousMetricLogElement>> const&, unsigned long) @ 0x00000000182d69bb in /usr/bin/clickhouse
2023.11.15 02:38:57.010348 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 766.09 MiB, peak 766.09 MiB, free memory in arenas 37.01 MiB, will set to 791.09 MiB (RSS), difference: 25.00 MiB
2023.11.15 02:38:57.299522 [ 276 ] {} <Fatal> BaseDaemon: 21. /home/bharatnc/Code/ClickHouse/src/Interpreters/SystemLog.cpp:456: DB::SystemLog<DB::AsynchronousMetricLogElement>::savingThreadFunction() @ 0x00000000182d596e in /usr/bin/clickhouse
2023.11.15 02:38:57.695550 [ 276 ] {} <Fatal> BaseDaemon: 22. /home/bharatnc/Code/ClickHouse/src/Common/SystemLogBase.cpp:248: DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()::operator()() const @ 0x000000000ff42099 in /usr/bin/clickhouse
2023.11.15 02:38:58.003904 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 791.09 MiB, peak 791.09 MiB, free memory in arenas 36.96 MiB, will set to 793.47 MiB (RSS), difference: 2.38 MiB
2023.11.15 02:38:58.097489 [ 276 ] {} <Fatal> BaseDaemon: 23. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/invoke.h:394: decltype(std::declval<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&>()()) std::__invoke[abi:v15000]<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&) @ 0x000000000ff42075 in /usr/bin/clickhouse
2023.11.15 02:38:58.491747 [ 276 ] {} <Fatal> BaseDaemon: 24. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/tuple:1789: decltype(auto) std::__apply_tuple_impl[abi:v15000]<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&, std::tuple<>&>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&, std::tuple<>&, std::__tuple_indices<>) @ 0x000000000ff42059 in /usr/bin/clickhouse
2023.11.15 02:38:58.885425 [ 276 ] {} <Fatal> BaseDaemon: 25. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/tuple:1798: decltype(auto) std::apply[abi:v15000]<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&, std::tuple<>&>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&, std::tuple<>&) @ 0x000000000ff41fbd in /usr/bin/clickhouse
2023.11.15 02:38:59.179119 [ 276 ] {} <Fatal> BaseDaemon: 26. /home/bharatnc/Code/ClickHouse/src/Common/ThreadPool.h:242: ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&)::'lambda'()::operator()() @ 0x000000000ff41ee5 in /usr/bin/clickhouse
2023.11.15 02:38:59.555881 [ 276 ] {} <Fatal> BaseDaemon: 27. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/invoke.h:394: decltype(std::declval<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>()()) std::__invoke[abi:v15000]<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&)::'lambda'()&>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&) @ 0x000000000ff41e55 in /usr/bin/clickhouse
2023.11.15 02:38:59.816028 [ 249 ] {} <Trace> SystemLog (system.trace_log): Flushing system log, 15 entries to flush up to offset 33
2023.11.15 02:38:59.839723 [ 249 ] {} <Trace> system.trace_log (a186b92a-b8e4-4174-9a96-1fcf2169a16b): Trying to reserve 1.00 MiB using storage policy from min volume index 0
2023.11.15 02:38:59.839910 [ 249 ] {} <Trace> DiskLocal: Reserved 1.00 MiB on local disk `default`, having unreserved 585.69 GiB.
2023.11.15 02:38:59.849907 [ 249 ] {} <Trace> MergedBlockOutputStream: filled checksums 202311_2_2_0 (state Temporary)
2023.11.15 02:38:59.861063 [ 249 ] {} <Trace> system.trace_log (a186b92a-b8e4-4174-9a96-1fcf2169a16b): Renaming temporary part tmp_insert_202311_2_2_0 to 202311_2_2_0 with tid (1, 1, 00000000-0000-0000-0000-000000000000).
2023.11.15 02:38:59.861528 [ 249 ] {} <Test> system.trace_log (a186b92a-b8e4-4174-9a96-1fcf2169a16b): preparePartForCommit: inserting 202311_2_2_0 (state PreActive) into data_parts_indexes
2023.11.15 02:38:59.865293 [ 249 ] {} <Trace> SystemLog (system.trace_log): Flushed system log up to offset 33
2023.11.15 02:38:59.915113 [ 276 ] {} <Fatal> BaseDaemon: 28. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/invoke.h:480: void std::__invoke_void_return_wrapper<void, true>::__call<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&)::'lambda'()&>(ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&)::'lambda'()&) @ 0x000000000ff41e35 in /usr/bin/clickhouse
2023.11.15 02:39:00.003627 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 793.98 MiB, peak 798.15 MiB, free memory in arenas 36.91 MiB, will set to 800.84 MiB (RSS), difference: 6.86 MiB
2023.11.15 02:39:00.282893 [ 276 ] {} <Fatal> BaseDaemon: 29. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/function.h:235: std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&)::'lambda'(), void ()>::operator()[abi:v15000]() @ 0x000000000ff41e15 in /usr/bin/clickhouse
2023.11.15 02:39:00.640841 [ 276 ] {} <Fatal> BaseDaemon: 30. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/function.h:716: void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0x000000000ff41de0 in /usr/bin/clickhouse
2023.11.15 02:39:00.706994 [ 276 ] {} <Fatal> BaseDaemon: 31. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/function.h:848: std::__function::__policy_func<void ()>::operator()[abi:v15000]() const @ 0x000000000fdbddf6 in /usr/bin/clickhouse
2023.11.15 02:39:00.768623 [ 276 ] {} <Fatal> BaseDaemon: 32. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/function.h:1187: std::function<void ()>::operator()() const @ 0x000000000fdbd2d5 in /usr/bin/clickhouse
2023.11.15 02:39:00.856958 [ 276 ] {} <Fatal> BaseDaemon: 33. /home/bharatnc/Code/ClickHouse/src/Common/ThreadPool.cpp:426: ThreadPoolImpl<std::thread>::worker(std::__list_iterator<std::thread, void*>) @ 0x000000000fec5a23 in /usr/bin/clickhouse
2023.11.15 02:39:00.963792 [ 276 ] {} <Fatal> BaseDaemon: 34. /home/bharatnc/Code/ClickHouse/src/Common/ThreadPool.cpp:179: void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()::operator()() const @ 0x000000000fecdcc4 in /usr/bin/clickhouse
2023.11.15 02:39:01.003603 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 800.84 MiB, peak 800.84 MiB, free memory in arenas 36.87 MiB, will set to 802.72 MiB (RSS), difference: 1.88 MiB
2023.11.15 02:39:01.082294 [ 276 ] {} <Fatal> BaseDaemon: 35. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/invoke.h:394: decltype(std::declval<void>()()) std::__invoke[abi:v15000]<void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>(void&&) @ 0x000000000fecdc75 in /usr/bin/clickhouse
2023.11.15 02:39:01.196916 [ 276 ] {} <Fatal> BaseDaemon: 36. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/thread:285: void std::__thread_execute[abi:v15000]<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>(std::tuple<void, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>&, std::__tuple_indices<>) @ 0x000000000fecdb9d in /usr/bin/clickhouse
2023.11.15 02:39:01.311951 [ 276 ] {} <Fatal> BaseDaemon: 37. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/thread:295: void* std::__thread_proxy[abi:v15000]<std::tuple<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>>(void*) @ 0x000000000fecd682 in /usr/bin/clickhouse
2023.11.15 02:39:01.312090 [ 276 ] {} <Fatal> BaseDaemon: 38. ? @ 0x00007f6970d96ac3 in ?
2023.11.15 02:39:01.312169 [ 276 ] {} <Fatal> BaseDaemon: 39. ? @ 0x00007f6970e28a40 in ?
2023.11.15 02:39:01.312254 [ 276 ] {} <Fatal> BaseDaemon: Integrity check of the executable skipped because the reference checksum could not be read.
2023.11.15 02:39:01.937662 [ 250 ] {} <Trace> SystemLog (system.metric_log): Flushing system log, 9 entries to flush up to offset 14
2023.11.15 02:39:02.742983 [ 34 ] {} <Debug> DNSResolver: Updating DNS cache
2023.11.15 02:39:02.744943 [ 34 ] {} <Debug> DNSResolver: Updated DNS cache

```

Reply:
Now, that I look at the logs more closely, there is actually a segmentation fault that follows the above failure. Here's the entire log:
```
2023.11.15 02:38:54.309088 [ 250 ] {} <Test> system.metric_log (136bdb91-0233-4d7c-a201-fdb1ba9598c3): preparePartForCommit: inserting 202311_1_1_0 (state PreActive) into data_parts_indexes
2023.11.15 02:38:54.437301 [ 250 ] {} <Trace> SystemLog (system.metric_log): Flushed system log up to offset 5
2023.11.15 02:38:54.437359 [ 8 ] {37093498-37dd-4d59-a63b-b2c89b0a74b4} <Debug> SystemLogQueue (system.asynchronous_metric_log): Requested flush up to offset 2360
2023.11.15 02:38:54.437484 [ 252 ] {} <Trace> SystemLog (system.asynchronous_metric_log): Flushing system log, 2360 entries to flush up to offset 2360
2023.11.15 02:38:54.437732 [ 252 ] {} <Debug> SystemLog (system.asynchronous_metric_log): Creating new table system.asynchronous_metric_log for AsynchronousMetricLog
2023.11.15 02:38:54.468529 [ 252 ] {} <Debug> system.asynchronous_metric_log (6769929f-56b8-41b5-a60b-c766a01465a6): Loading data parts
2023.11.15 02:38:54.468983 [ 252 ] {} <Test> system.asynchronous_metric_log (6769929f-56b8-41b5-a60b-c766a01465a6): loadDataParts: clearing data_parts_indexes (had 0 parts)
2023.11.15 02:38:54.469096 [ 252 ] {} <Debug> system.asynchronous_metric_log (6769929f-56b8-41b5-a60b-c766a01465a6): There are no data parts
2023.11.15 02:38:54.545868 [ 252 ] {} <Fatal> : Logical error: 'Invalid number of columns in chunk pushed to OutputPort. Expected 4, found 5
Header: event_date Date UInt16(size = 0), event_time DateTime UInt32(size = 0), metric LowCardinality(String) ColumnLowCardinality(size = 0, UInt8(size = 0), ColumnUnique(size = 1, String(size = 1))), value Float64 Float64(size = 0)
Chunk:  ColumnLowCardinality(size = 2360, UInt8(size = 2360), ColumnUnique(size = 2, String(size = 2))) UInt16(size = 2360) UInt32(size = 2360) ColumnLowCardinality(size = 2360, UInt16(size = 2360), ColumnUnique(size = 374, String(size = 374))) Float64(size = 2360)
'.
2023.11.15 02:38:54.546358 [ 7 ] {} <Trace> BaseDaemon: Received signal 6
2023.11.15 02:38:54.547176 [ 276 ] {} <Fatal> BaseDaemon: ########## Short fault info ############
2023.11.15 02:38:54.547394 [ 276 ] {} <Fatal> BaseDaemon: (version 23.11.1.1, build id: 1F7F5CA22F8163637736DBB9DDCF5E1352AB160A, git hash: 4f677132514edae42ebe538ad242f38d73ac74c5) (from thread 252) Received signal 6
2023.11.15 02:38:54.547518 [ 276 ] {} <Fatal> BaseDaemon: Signal description: Aborted
2023.11.15 02:38:54.547599 [ 276 ] {} <Fatal> BaseDaemon: 
2023.11.15 02:38:54.547736 [ 276 ] {} <Fatal> BaseDaemon: Stack trace: 0x00007f6970d989fc 0x00007f6970d44476 0x00007f6970d2a7f3 0x000000000fd5017f 0x000000000fd50215 0x000000000fd506d1 0x0000000007c63fea 0x00000000184c7984 0x0000000019967303 0x00000000199afdd9 0x00000000100964be 0x000000001998e6fc 0x000000001998511f 0x0000000019984b7e 0x00000000199aeeeb 0x00000000199aefd9 0x00000000182d69bb 0x00000000182d596e 0x000000000ff42099 0x000000000ff42075 0x000000000ff42059 0x000000000ff41fbd 0x000000000ff41ee5 0x000000000ff41e55 0x000000000ff41e35 0x000000000ff41e15 0x000000000ff41de0 0x000000000fdbddf6 0x000000000fdbd2d5 0x000000000fec5a23 0x000000000fecdcc4 0x000000000fecdc75 0x000000000fecdb9d 0x000000000fecd682 0x00007f6970d96ac3 0x00007f6970e28a40
2023.11.15 02:38:54.547846 [ 276 ] {} <Fatal> BaseDaemon: ########################################
2023.11.15 02:38:54.547918 [ 276 ] {} <Fatal> BaseDaemon: (version 23.11.1.1, build id: 1F7F5CA22F8163637736DBB9DDCF5E1352AB160A, git hash: 4f677132514edae42ebe538ad242f38d73ac74c5) (from thread 252) (no query) Received signal Aborted (6)
2023.11.15 02:38:54.548017 [ 276 ] {} <Fatal> BaseDaemon: 
2023.11.15 02:38:54.548078 [ 276 ] {} <Fatal> BaseDaemon: Stack trace: 0x00007f6970d989fc 0x00007f6970d44476 0x00007f6970d2a7f3 0x000000000fd5017f 0x000000000fd50215 0x000000000fd506d1 0x0000000007c63fea 0x00000000184c7984 0x0000000019967303 0x00000000199afdd9 0x00000000100964be 0x000000001998e6fc 0x000000001998511f 0x0000000019984b7e 0x00000000199aeeeb 0x00000000199aefd9 0x00000000182d69bb 0x00000000182d596e 0x000000000ff42099 0x000000000ff42075 0x000000000ff42059 0x000000000ff41fbd 0x000000000ff41ee5 0x000000000ff41e55 0x000000000ff41e35 0x000000000ff41e15 0x000000000ff41de0 0x000000000fdbddf6 0x000000000fdbd2d5 0x000000000fec5a23 0x000000000fecdcc4 0x000000000fecdc75 0x000000000fecdb9d 0x000000000fecd682 0x00007f6970d96ac3 0x00007f6970e28a40
2023.11.15 02:38:54.548206 [ 276 ] {} <Fatal> BaseDaemon: 4. ? @ 0x00007f6970d989fc in ?
2023.11.15 02:38:54.548300 [ 276 ] {} <Fatal> BaseDaemon: 5. ? @ 0x00007f6970d44476 in ?
2023.11.15 02:38:54.548373 [ 276 ] {} <Fatal> BaseDaemon: 6. ? @ 0x00007f6970d2a7f3 in ?
2023.11.15 02:38:54.667140 [ 276 ] {} <Fatal> BaseDaemon: 7. /home/bharatnc/Code/ClickHouse/src/Common/Exception.cpp:0: DB::abortOnFailedAssertion(String const&) @ 0x000000000fd5017f in /usr/bin/clickhouse
2023.11.15 02:38:54.782152 [ 276 ] {} <Fatal> BaseDaemon: 8. /home/bharatnc/Code/ClickHouse/src/Common/Exception.cpp:64: DB::handle_error_code(String const&, int, bool, std::vector<void*, std::allocator<void*>> const&) @ 0x000000000fd50215 in /usr/bin/clickhouse
2023.11.15 02:38:54.882198 [ 276 ] {} <Fatal> BaseDaemon: 9. /home/bharatnc/Code/ClickHouse/src/Common/Exception.cpp:101: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000fd506d1 in /usr/bin/clickhouse
2023.11.15 02:38:54.992105 [ 86 ] {} <Trace> test.test_table (fabdf8d3-2cf6-4ba5-b8aa-b60fdfa43437): Scheduling next merge selecting task after 7802ms
2023.11.15 02:38:54.996870 [ 276 ] {} <Fatal> BaseDaemon: 10. /home/bharatnc/Code/ClickHouse/src/Common/Exception.h:85: DB::Exception::Exception(String&&, int, bool) @ 0x0000000007c63fea in /usr/bin/clickhouse
2023.11.15 02:38:55.002621 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 588.16 MiB, peak 596.97 MiB, free memory in arenas 37.23 MiB, will set to 656.97 MiB (RSS), difference: 68.80 MiB
2023.11.15 02:38:55.312126 [ 276 ] {} <Fatal> BaseDaemon: 11. /home/bharatnc/Code/ClickHouse/src/Common/Exception.h:113: DB::Exception::Exception<unsigned long, unsigned long, String, String>(int, FormatStringHelperImpl<std::type_identity<unsigned long>::type, std::type_identity<unsigned long>::type, std::type_identity<String>::type, std::type_identity<String>::type>, unsigned long&&, unsigned long&&, String&&, String&&) @ 0x00000000184c7984 in /usr/bin/clickhouse
2023.11.15 02:38:55.399169 [ 276 ] {} <Fatal> BaseDaemon: 12.1. inlined from /home/bharatnc/Code/ClickHouse/src/Processors/Port.h:415: DB::OutputPort::pushData(DB::Port::State::Data)
2023.11.15 02:38:55.399326 [ 276 ] {} <Fatal> BaseDaemon: 12. /home/bharatnc/Code/ClickHouse/src/Processors/ISource.cpp:40: DB::ISource::prepare() @ 0x0000000019967303 in /usr/bin/clickhouse
2023.11.15 02:38:55.484314 [ 276 ] {} <Fatal> BaseDaemon: 13. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PushingPipelineExecutor.cpp:36: DB::PushingSource::prepare() @ 0x00000000199afdd9 in /usr/bin/clickhouse
2023.11.15 02:38:56.004014 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 656.97 MiB, peak 656.97 MiB, free memory in arenas 37.13 MiB, will set to 766.09 MiB (RSS), difference: 109.13 MiB
2023.11.15 02:38:56.023950 [ 276 ] {} <Fatal> BaseDaemon: 14. /home/bharatnc/Code/ClickHouse/src/Processors/IProcessor.h:193: DB::IProcessor::prepare(std::vector<unsigned long, std::allocator<unsigned long>> const&, std::vector<unsigned long, std::allocator<unsigned long>> const&) @ 0x00000000100964be in /usr/bin/clickhouse
2023.11.15 02:38:56.149883 [ 276 ] {} <Fatal> BaseDaemon: 15. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/ExecutingGraph.cpp:283: DB::ExecutingGraph::updateNode(unsigned long, std::queue<DB::ExecutingGraph::Node*, std::deque<DB::ExecutingGraph::Node*, std::allocator<DB::ExecutingGraph::Node*>>>&, std::queue<DB::ExecutingGraph::Node*, std::deque<DB::ExecutingGraph::Node*, std::allocator<DB::ExecutingGraph::Node*>>>&) @ 0x000000001998e6fc in /usr/bin/clickhouse
2023.11.15 02:38:56.286972 [ 276 ] {} <Fatal> BaseDaemon: 16. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PipelineExecutor.cpp:291: DB::PipelineExecutor::executeStepImpl(unsigned long, std::atomic<bool>*) @ 0x000000001998511f in /usr/bin/clickhouse
2023.11.15 02:38:56.418039 [ 276 ] {} <Fatal> BaseDaemon: 17. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PipelineExecutor.cpp:147: DB::PipelineExecutor::executeStep(std::atomic<bool>*) @ 0x0000000019984b7e in /usr/bin/clickhouse
2023.11.15 02:38:56.499182 [ 276 ] {} <Fatal> BaseDaemon: 18. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PushingPipelineExecutor.cpp:105: DB::PushingPipelineExecutor::push(DB::Chunk) @ 0x00000000199aeeeb in /usr/bin/clickhouse
2023.11.15 02:38:56.580706 [ 276 ] {} <Fatal> BaseDaemon: 19. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PushingPipelineExecutor.cpp:112: DB::PushingPipelineExecutor::push(DB::Block) @ 0x00000000199aefd9 in /usr/bin/clickhouse
2023.11.15 02:38:56.940844 [ 276 ] {} <Fatal> BaseDaemon: 20. /home/bharatnc/Code/ClickHouse/src/Interpreters/SystemLog.cpp:515: DB::SystemLog<DB::AsynchronousMetricLogElement>::flushImpl(std::vector<DB::AsynchronousMetricLogElement, std::allocator<DB::AsynchronousMetricLogElement>> const&, unsigned long) @ 0x00000000182d69bb in /usr/bin/clickhouse
2023.11.15 02:38:57.010348 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 766.09 MiB, peak 766.09 MiB, free memory in arenas 37.01 MiB, will set to 791.09 MiB (RSS), difference: 25.00 MiB
2023.11.15 02:38:57.299522 [ 276 ] {} <Fatal> BaseDaemon: 21. /home/bharatnc/Code/ClickHouse/src/Interpreters/SystemLog.cpp:456: DB::SystemLog<DB::AsynchronousMetricLogElement>::savingThreadFunction() @ 0x00000000182d596e in /usr/bin/clickhouse
2023.11.15 02:38:57.695550 [ 276 ] {} <Fatal> BaseDaemon: 22. /home/bharatnc/Code/ClickHouse/src/Common/SystemLogBase.cpp:248: DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()::operator()() const @ 0x000000000ff42099 in /usr/bin/clickhouse
2023.11.15 02:38:58.003904 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 791.09 MiB, peak 791.09 MiB, free memory in arenas 36.96 MiB, will set to 793.47 MiB (RSS), difference: 2.38 MiB
2023.11.15 02:38:58.097489 [ 276 ] {} <Fatal> BaseDaemon: 23. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/invoke.h:394: decltype(std::declval<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&>()()) std::__invoke[abi:v15000]<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&) @ 0x000000000ff42075 in /usr/bin/clickhouse
2023.11.15 02:38:58.491747 [ 276 ] {} <Fatal> BaseDaemon: 24. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/tuple:1789: decltype(auto) std::__apply_tuple_impl[abi:v15000]<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&, std::tuple<>&>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&, std::tuple<>&, std::__tuple_indices<>) @ 0x000000000ff42059 in /usr/bin/clickhouse
2023.11.15 02:38:58.885425 [ 276 ] {} <Fatal> BaseDaemon: 25. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/tuple:1798: decltype(auto) std::apply[abi:v15000]<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&, std::tuple<>&>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&, std::tuple<>&) @ 0x000000000ff41fbd in /usr/bin/clickhouse
2023.11.15 02:38:59.179119 [ 276 ] {} <Fatal> BaseDaemon: 26. /home/bharatnc/Code/ClickHouse/src/Common/ThreadPool.h:242: ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&)::'lambda'()::operator()() @ 0x000000000ff41ee5 in /usr/bin/clickhouse
2023.11.15 02:38:59.555881 [ 276 ] {} <Fatal> BaseDaemon: 27. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/invoke.h:394: decltype(std::declval<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>()()) std::__invoke[abi:v15000]<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&)::'lambda'()&>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&) @ 0x000000000ff41e55 in /usr/bin/clickhouse
2023.11.15 02:38:59.816028 [ 249 ] {} <Trace> SystemLog (system.trace_log): Flushing system log, 15 entries to flush up to offset 33
2023.11.15 02:38:59.839723 [ 249 ] {} <Trace> system.trace_log (a186b92a-b8e4-4174-9a96-1fcf2169a16b): Trying to reserve 1.00 MiB using storage policy from min volume index 0
2023.11.15 02:38:59.839910 [ 249 ] {} <Trace> DiskLocal: Reserved 1.00 MiB on local disk `default`, having unreserved 585.69 GiB.
2023.11.15 02:38:59.849907 [ 249 ] {} <Trace> MergedBlockOutputStream: filled checksums 202311_2_2_0 (state Temporary)
2023.11.15 02:38:59.861063 [ 249 ] {} <Trace> system.trace_log (a186b92a-b8e4-4174-9a96-1fcf2169a16b): Renaming temporary part tmp_insert_202311_2_2_0 to 202311_2_2_0 with tid (1, 1, 00000000-0000-0000-0000-000000000000).
2023.11.15 02:38:59.861528 [ 249 ] {} <Test> system.trace_log (a186b92a-b8e4-4174-9a96-1fcf2169a16b): preparePartForCommit: inserting 202311_2_2_0 (state PreActive) into data_parts_indexes
2023.11.15 02:38:59.865293 [ 249 ] {} <Trace> SystemLog (system.trace_log): Flushed system log up to offset 33
2023.11.15 02:38:59.915113 [ 276 ] {} <Fatal> BaseDaemon: 28. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/invoke.h:480: void std::__invoke_void_return_wrapper<void, true>::__call<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&)::'lambda'()&>(ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&)::'lambda'()&) @ 0x000000000ff41e35 in /usr/bin/clickhouse
2023.11.15 02:39:00.003627 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 793.98 MiB, peak 798.15 MiB, free memory in arenas 36.91 MiB, will set to 800.84 MiB (RSS), difference: 6.86 MiB
2023.11.15 02:39:00.282893 [ 276 ] {} <Fatal> BaseDaemon: 29. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/function.h:235: std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&)::'lambda'(), void ()>::operator()[abi:v15000]() @ 0x000000000ff41e15 in /usr/bin/clickhouse
2023.11.15 02:39:00.640841 [ 276 ] {} <Fatal> BaseDaemon: 30. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/function.h:716: void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()>(DB::SystemLogBase<DB::AsynchronousMetricLogElement>::startup()::'lambda'()&&)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0x000000000ff41de0 in /usr/bin/clickhouse
2023.11.15 02:39:00.706994 [ 276 ] {} <Fatal> BaseDaemon: 31. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/function.h:848: std::__function::__policy_func<void ()>::operator()[abi:v15000]() const @ 0x000000000fdbddf6 in /usr/bin/clickhouse
2023.11.15 02:39:00.768623 [ 276 ] {} <Fatal> BaseDaemon: 32. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/function.h:1187: std::function<void ()>::operator()() const @ 0x000000000fdbd2d5 in /usr/bin/clickhouse
2023.11.15 02:39:00.856958 [ 276 ] {} <Fatal> BaseDaemon: 33. /home/bharatnc/Code/ClickHouse/src/Common/ThreadPool.cpp:426: ThreadPoolImpl<std::thread>::worker(std::__list_iterator<std::thread, void*>) @ 0x000000000fec5a23 in /usr/bin/clickhouse
2023.11.15 02:39:00.963792 [ 276 ] {} <Fatal> BaseDaemon: 34. /home/bharatnc/Code/ClickHouse/src/Common/ThreadPool.cpp:179: void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()::operator()() const @ 0x000000000fecdcc4 in /usr/bin/clickhouse
2023.11.15 02:39:01.003603 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 800.84 MiB, peak 800.84 MiB, free memory in arenas 36.87 MiB, will set to 802.72 MiB (RSS), difference: 1.88 MiB
2023.11.15 02:39:01.082294 [ 276 ] {} <Fatal> BaseDaemon: 35. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/__functional/invoke.h:394: decltype(std::declval<void>()()) std::__invoke[abi:v15000]<void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>(void&&) @ 0x000000000fecdc75 in /usr/bin/clickhouse
2023.11.15 02:39:01.196916 [ 276 ] {} <Fatal> BaseDaemon: 36. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/thread:285: void std::__thread_execute[abi:v15000]<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>(std::tuple<void, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>&, std::__tuple_indices<>) @ 0x000000000fecdb9d in /usr/bin/clickhouse
2023.11.15 02:39:01.311951 [ 276 ] {} <Fatal> BaseDaemon: 37. /home/bharatnc/Code/ClickHouse/contrib/llvm-project/libcxx/include/thread:295: void* std::__thread_proxy[abi:v15000]<std::tuple<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>>(void*) @ 0x000000000fecd682 in /usr/bin/clickhouse
2023.11.15 02:39:01.312090 [ 276 ] {} <Fatal> BaseDaemon: 38. ? @ 0x00007f6970d96ac3 in ?
2023.11.15 02:39:01.312169 [ 276 ] {} <Fatal> BaseDaemon: 39. ? @ 0x00007f6970e28a40 in ?
2023.11.15 02:39:01.312254 [ 276 ] {} <Fatal> BaseDaemon: Integrity check of the executable skipped because the reference checksum could not be read.
2023.11.15 02:39:01.937662 [ 250 ] {} <Trace> SystemLog (system.metric_log): Flushing system log, 9 entries to flush up to offset 14
2023.11.15 02:39:02.742983 [ 34 ] {} <Debug> DNSResolver: Updating DNS cache
2023.11.15 02:39:02.744943 [ 34 ] {} <Debug> DNSResolver: Updated DNS cache
2023.11.15 02:39:02.798879 [ 11 ] {} <Trace> test.test_table (fabdf8d3-2cf6-4ba5-b8aa-b60fdfa43437): Scheduling next merge selecting task after 9935ms
2023.11.15 02:39:02.821022 [ 250 ] {} <Trace> system.metric_log (136bdb91-0233-4d7c-a201-fdb1ba9598c3): Trying to reserve 1.00 MiB using storage policy from min volume index 0
2023.11.15 02:39:02.821172 [ 250 ] {} <Trace> DiskLocal: Reserved 1.00 MiB on local disk `default`, having unreserved 585.69 GiB.
2023.11.15 02:39:03.111936 [ 250 ] {} <Trace> MergedBlockOutputStream: filled checksums 202311_2_2_0 (state Temporary)
2023.11.15 02:39:03.504769 [ 250 ] {} <Trace> system.metric_log (136bdb91-0233-4d7c-a201-fdb1ba9598c3): Renaming temporary part tmp_insert_202311_2_2_0 to 202311_2_2_0 with tid (1, 1, 00000000-0000-0000-0000-000000000000).
2023.11.15 02:39:03.505221 [ 250 ] {} <Test> system.metric_log (136bdb91-0233-4d7c-a201-fdb1ba9598c3): preparePartForCommit: inserting 202311_2_2_0 (state PreActive) into data_parts_indexes
2023.11.15 02:39:03.652675 [ 250 ] {} <Trace> SystemLog (system.metric_log): Flushed system log up to offset 14
2023.11.15 02:39:04.002747 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 804.51 MiB, peak 812.51 MiB, free memory in arenas 29.15 MiB, will set to 811.02 MiB (RSS), difference: 6.52 MiB
2023.11.15 02:39:07.365613 [ 249 ] {} <Trace> SystemLog (system.trace_log): Flushing system log, 14 entries to flush up to offset 47
2023.11.15 02:39:07.378679 [ 249 ] {} <Trace> system.trace_log (a186b92a-b8e4-4174-9a96-1fcf2169a16b): Trying to reserve 1.00 MiB using storage policy from min volume index 0
2023.11.15 02:39:07.378815 [ 249 ] {} <Trace> DiskLocal: Reserved 1.00 MiB on local disk `default`, having unreserved 585.68 GiB.
2023.11.15 02:39:07.384825 [ 249 ] {} <Trace> MergedBlockOutputStream: filled checksums 202311_3_3_0 (state Temporary)
2023.11.15 02:39:07.392651 [ 249 ] {} <Trace> system.trace_log (a186b92a-b8e4-4174-9a96-1fcf2169a16b): Renaming temporary part tmp_insert_202311_3_3_0 to 202311_3_3_0 with tid (1, 1, 00000000-0000-0000-0000-000000000000).
2023.11.15 02:39:07.393039 [ 249 ] {} <Test> system.trace_log (a186b92a-b8e4-4174-9a96-1fcf2169a16b): preparePartForCommit: inserting 202311_3_3_0 (state PreActive) into data_parts_indexes
2023.11.15 02:39:07.396533 [ 249 ] {} <Trace> SystemLog (system.trace_log): Flushed system log up to offset 47
2023.11.15 02:39:08.106181 [ 276 ] {} <Debug> SystemLogQueue (system.crash_log): Requested flush up to offset 1
2023.11.15 02:39:08.106227 [ 253 ] {} <Trace> SystemLog (system.crash_log): Flushing system log, 1 entries to flush up to offset 1
2023.11.15 02:39:08.106310 [ 276 ] {} <Information> SentryWriter: Not sending crash report
2023.11.15 02:39:08.106376 [ 276 ] {} <Fatal> BaseDaemon: This ClickHouse version is not official and should be upgraded to the official build.
2023.11.15 02:39:08.124589 [ 253 ] {} <Trace> system.crash_log (ba89acee-93ce-4ad6-a436-4952f3651942): Trying to reserve 1.00 MiB using storage policy from min volume index 0
2023.11.15 02:39:08.124768 [ 253 ] {} <Trace> DiskLocal: Reserved 1.00 MiB on local disk `default`, having unreserved 585.68 GiB.
2023.11.15 02:39:08.131805 [ 253 ] {} <Trace> MergedBlockOutputStream: filled checksums all_1_1_0 (state Temporary)
2023.11.15 02:39:08.139644 [ 253 ] {} <Trace> system.crash_log (ba89acee-93ce-4ad6-a436-4952f3651942): Renaming temporary part tmp_insert_all_1_1_0 to all_1_1_0 with tid (1, 1, 00000000-0000-0000-0000-000000000000).
2023.11.15 02:39:08.140081 [ 253 ] {} <Test> system.crash_log (ba89acee-93ce-4ad6-a436-4952f3651942): preparePartForCommit: inserting all_1_1_0 (state PreActive) into data_parts_indexes
2023.11.15 02:39:08.143420 [ 253 ] {} <Trace> SystemLog (system.crash_log): Flushed system log up to offset 1
2023.11.15 02:39:11.153032 [ 250 ] {} <Trace> SystemLog (system.metric_log): Flushing system log, 10 entries to flush up to offset 24
2023.11.15 02:39:11.547569 [ 7 ] {} <Trace> BaseDaemon: Received signal 11
2023.11.15 02:39:11.548058 [ 277 ] {} <Fatal> BaseDaemon: ########## Short fault info ############
2023.11.15 02:39:11.548213 [ 277 ] {} <Fatal> BaseDaemon: (version 23.11.1.1, build id: 1F7F5CA22F8163637736DBB9DDCF5E1352AB160A, git hash: 4f677132514edae42ebe538ad242f38d73ac74c5) (from thread 252) Received signal 11
2023.11.15 02:39:11.548338 [ 277 ] {} <Fatal> BaseDaemon: Signal description: Segmentation fault
2023.11.15 02:39:11.548415 [ 277 ] {} <Fatal> BaseDaemon: Address: NULL pointer. Access: read. Unknown si_code.
2023.11.15 02:39:11.548546 [ 277 ] {} <Fatal> BaseDaemon: Stack trace: 0x00007f6970d2a898 0x000000000fd5017f 0x000000000fd50215 0x000000000fd506d1 0x0000000007c63fea 0x00000000184c7984 0x0000000019967303 0x00000000199afdd9 0x00000000100964be 0x000000001998e6fc 0x000000001998511f 0x0000000019984b7e 0x00000000199aeeeb 0x00000000199aefd9 0x00000000182d69bb 0x00000000182d596e 0x000000000ff42099 0x000000000ff42075 0x000000000ff42059 0x000000000ff41fbd 0x000000000ff41ee5 0x000000000ff41e55 0x000000000ff41e35 0x000000000ff41e15 0x000000000ff41de0 0x000000000fdbddf6 0x000000000fdbd2d5 0x000000000fec5a23 0x000000000fecdcc4 0x000000000fecdc75 0x000000000fecdb9d 0x000000000fecd682 0x00007f6970d96ac3 0x00007f6970e28a40
2023.11.15 02:39:11.548657 [ 277 ] {} <Fatal> BaseDaemon: ########################################
2023.11.15 02:39:11.548730 [ 277 ] {} <Fatal> BaseDaemon: (version 23.11.1.1, build id: 1F7F5CA22F8163637736DBB9DDCF5E1352AB160A, git hash: 4f677132514edae42ebe538ad242f38d73ac74c5) (from thread 252) (no query) Received signal Segmentation fault (11)
2023.11.15 02:39:11.548829 [ 277 ] {} <Fatal> BaseDaemon: Address: NULL pointer. Access: read. Unknown si_code.
2023.11.15 02:39:11.548903 [ 277 ] {} <Fatal> BaseDaemon: Stack trace: 0x00007f6970d2a898 0x000000000fd5017f 0x000000000fd50215 0x000000000fd506d1 0x0000000007c63fea 0x00000000184c7984 0x0000000019967303 0x00000000199afdd9 0x00000000100964be 0x000000001998e6fc 0x000000001998511f 0x0000000019984b7e 0x00000000199aeeeb 0x00000000199aefd9 0x00000000182d69bb 0x00000000182d596e 0x000000000ff42099 0x000000000ff42075 0x000000000ff42059 0x000000000ff41fbd 0x000000000ff41ee5 0x000000000ff41e55 0x000000000ff41e35 0x000000000ff41e15 0x000000000ff41de0 0x000000000fdbddf6 0x000000000fdbd2d5 0x000000000fec5a23 0x000000000fecdcc4 0x000000000fecdc75 0x000000000fecdb9d 0x000000000fecd682 0x00007f6970d96ac3 0x00007f6970e28a40
2023.11.15 02:39:11.549038 [ 277 ] {} <Fatal> BaseDaemon: 4. ? @ 0x00007f6970d2a898 in ?
2023.11.15 02:39:11.666456 [ 277 ] {} <Fatal> BaseDaemon: 5. /home/bharatnc/Code/ClickHouse/src/Common/Exception.cpp:0: DB::abortOnFailedAssertion(String const&) @ 0x000000000fd5017f in /usr/bin/clickhouse
2023.11.15 02:39:11.785523 [ 277 ] {} <Fatal> BaseDaemon: 6. /home/bharatnc/Code/ClickHouse/src/Common/Exception.cpp:64: DB::handle_error_code(String const&, int, bool, std::vector<void*, std::allocator<void*>> const&) @ 0x000000000fd50215 in /usr/bin/clickhouse
2023.11.15 02:39:11.852875 [ 250 ] {} <Trace> system.metric_log (136bdb91-0233-4d7c-a201-fdb1ba9598c3): Trying to reserve 1.00 MiB using storage policy from min volume index 0
2023.11.15 02:39:11.853042 [ 250 ] {} <Trace> DiskLocal: Reserved 1.00 MiB on local disk `default`, having unreserved 585.68 GiB.
2023.11.15 02:39:11.888740 [ 277 ] {} <Fatal> BaseDaemon: 7. /home/bharatnc/Code/ClickHouse/src/Common/Exception.cpp:101: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000fd506d1 in /usr/bin/clickhouse
2023.11.15 02:39:12.003993 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 815.50 MiB, peak 816.19 MiB, free memory in arenas 15.81 MiB, will set to 806.82 MiB (RSS), difference: -8.68 MiB
2023.11.15 02:39:12.008687 [ 277 ] {} <Fatal> BaseDaemon: 8. /home/bharatnc/Code/ClickHouse/src/Common/Exception.h:85: DB::Exception::Exception(String&&, int, bool) @ 0x0000000007c63fea in /usr/bin/clickhouse
2023.11.15 02:39:12.132405 [ 250 ] {} <Trace> MergedBlockOutputStream: filled checksums 202311_3_3_0 (state Temporary)
2023.11.15 02:39:12.340741 [ 277 ] {} <Fatal> BaseDaemon: 9. /home/bharatnc/Code/ClickHouse/src/Common/Exception.h:113: DB::Exception::Exception<unsigned long, unsigned long, String, String>(int, FormatStringHelperImpl<std::type_identity<unsigned long>::type, std::type_identity<unsigned long>::type, std::type_identity<String>::type, std::type_identity<String>::type>, unsigned long&&, unsigned long&&, String&&, String&&) @ 0x00000000184c7984 in /usr/bin/clickhouse
2023.11.15 02:39:12.431408 [ 277 ] {} <Fatal> BaseDaemon: 10.1. inlined from /home/bharatnc/Code/ClickHouse/src/Processors/Port.h:415: DB::OutputPort::pushData(DB::Port::State::Data)
2023.11.15 02:39:12.431541 [ 277 ] {} <Fatal> BaseDaemon: 10. /home/bharatnc/Code/ClickHouse/src/Processors/ISource.cpp:40: DB::ISource::prepare() @ 0x0000000019967303 in /usr/bin/clickhouse
2023.11.15 02:39:12.513563 [ 277 ] {} <Fatal> BaseDaemon: 11. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PushingPipelineExecutor.cpp:36: DB::PushingSource::prepare() @ 0x00000000199afdd9 in /usr/bin/clickhouse
2023.11.15 02:39:12.519545 [ 250 ] {} <Trace> system.metric_log (136bdb91-0233-4d7c-a201-fdb1ba9598c3): Renaming temporary part tmp_insert_202311_3_3_0 to 202311_3_3_0 with tid (1, 1, 00000000-0000-0000-0000-000000000000).
2023.11.15 02:39:12.520014 [ 250 ] {} <Test> system.metric_log (136bdb91-0233-4d7c-a201-fdb1ba9598c3): preparePartForCommit: inserting 202311_3_3_0 (state PreActive) into data_parts_indexes
2023.11.15 02:39:12.656408 [ 250 ] {} <Trace> SystemLog (system.metric_log): Flushed system log up to offset 24
2023.11.15 02:39:12.734407 [ 99 ] {} <Trace> test.test_table (fabdf8d3-2cf6-4ba5-b8aa-b60fdfa43437): Scheduling next merge selecting task after 11997ms
2023.11.15 02:39:13.002830 [ 262 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 803.54 MiB, peak 816.19 MiB, free memory in arenas 28.19 MiB, will set to 806.99 MiB (RSS), difference: 3.45 MiB
2023.11.15 02:39:13.012576 [ 277 ] {} <Fatal> BaseDaemon: 12. /home/bharatnc/Code/ClickHouse/src/Processors/IProcessor.h:193: DB::IProcessor::prepare(std::vector<unsigned long, std::allocator<unsigned long>> const&, std::vector<unsigned long, std::allocator<unsigned long>> const&) @ 0x00000000100964be in /usr/bin/clickhouse
2023.11.15 02:39:13.128559 [ 277 ] {} <Fatal> BaseDaemon: 13. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/ExecutingGraph.cpp:283: DB::ExecutingGraph::updateNode(unsigned long, std::queue<DB::ExecutingGraph::Node*, std::deque<DB::ExecutingGraph::Node*, std::allocator<DB::ExecutingGraph::Node*>>>&, std::queue<DB::ExecutingGraph::Node*, std::deque<DB::ExecutingGraph::Node*, std::allocator<DB::ExecutingGraph::Node*>>>&) @ 0x000000001998e6fc in /usr/bin/clickhouse
2023.11.15 02:39:13.258662 [ 277 ] {} <Fatal> BaseDaemon: 14. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PipelineExecutor.cpp:291: DB::PipelineExecutor::executeStepImpl(unsigned long, std::atomic<bool>*) @ 0x000000001998511f in /usr/bin/clickhouse
2023.11.15 02:39:13.389080 [ 277 ] {} <Fatal> BaseDaemon: 15. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PipelineExecutor.cpp:147: DB::PipelineExecutor::executeStep(std::atomic<bool>*) @ 0x0000000019984b7e in /usr/bin/clickhouse
2023.11.15 02:39:13.465781 [ 277 ] {} <Fatal> BaseDaemon: 16. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PushingPipelineExecutor.cpp:105: DB::PushingPipelineExecutor::push(DB::Chunk) @ 0x00000000199aeeeb in /usr/bin/clickhouse
2023.11.15 02:39:13.542638 [ 277 ] {} <Fatal> BaseDaemon: 17. /home/bharatnc/Code/ClickHouse/src/Processors/Executors/PushingPipelineExecutor.cpp:112: DB::PushingPipelineExecutor::push(DB::Block) @ 0x00000000199aefd9 in /usr/bin/clickhouse
2023.11.15 02:39:13.881464 [ 277 ] {} <Fatal> BaseDaemon: 18. /home/bharatnc/Code/ClickHouse/src/Interpreters/SystemLog.cpp:515: DB::SystemLog<DB::AsynchronousMetricLogElement>::flushImpl(std::vector<DB::AsynchronousMetricLogElement, std::allocator<DB::AsynchronousMetricLogElement>> const&, unsigned long) @ 0x00000000182d69bb in /usr/bin/clickhouse
2023.11.15 02:39:14.222393 [ 277 ] {} <Fatal> BaseDaemon: 19. /home/bharatnc/Code/ClickHouse/src/Interpreters/SystemLog.cpp:456: DB::SystemLog<DB::AsynchronousMetricLogElement>::savingThreadFunction() @ 0x00000000182d596e in /usr/bin/clickhouse
```

Reply:
The integration test `tests/integration/test_system_logs_hostname` that I'm introducing in this PR caught this crash. So those logs are actually from the integration test _instances.

Reply:
Builds look good now.